src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -348,6 +348,8 @@ object MatrixIR {
 abstract sealed class MatrixIR extends BaseIR {
   def typ: MatrixType
 
+  def partitionCounts: Option[Array[Long]] = None
```

Comment:
@tpoterba FYI I added (optional, static) partition counts to MatrixIR and they are used by VSM.count (if available).  They are written out by VSM.write and read back by read.  They should be preserved by map operations but invalided by filter operations, for example, but right now they're only available on MatrixRead.  They should also be computed by the OrderedRVD scan (more PRs coming).

tl;dr: read().count() should now be instantaneous.

Reply:
🚀